runtime/vm/createramclass.cpp

```diff
@@ -559,7 +569,14 @@ addInterfaceMethods(J9VMThread *vmStruct, J9ClassLoader *classLoader, J9Class *i
 										vTableMethodClassNameUTF = J9ROMCLASS_CLASSNAME(methodClass->romClass);
 									}
 									J9UTF8 *interfaceClassNameUTF = J9ROMCLASS_CLASSNAME(interfaceClass->romClass);
-									setClassLoadingConstraintOverrideError(vmStruct, J9ROMCLASS_CLASSNAME(romClass), vTableMethodLoader, vTableMethodClassNameUTF, interfaceLoader, interfaceClassNameUTF, interfaceClassNameUTF, J9UTF8_DATA(vTableMethodNameUTF), J9UTF8_LENGTH(vTableMethodNameUTF), J9UTF8_DATA(vTableMethodSigUTF), J9UTF8_LENGTH(vTableMethodSigUTF));
+									/* LinkageError will be thrown */
+									errorData->loader1 = vTableMethodLoader;
+									errorData->class1NameUTF = vTableMethodClassNameUTF;
+									errorData->loader2 = interfaceLoader;
+									errorData->class2NameUTF = interfaceClassNameUTF;
+									errorData->exceptionClassNameUTF = interfaceClassNameUTF;
+									errorData->methodNameUTF = vTableMethodNameUTF;
+									errorData->methodSigUTF = vTableMethodSigUTF;
```

Comment:
for code clarity, should this `goto fail` rather than `goto done`?

Reply:
Agreed - fixed.